src/elements/earn/portfolio/liquidityProtection/rewards/MyRewards.tsx

```diff
@@ -0,0 +1,108 @@
+import { useWeb3React } from '@web3-react/core';
+import { useEffect, useState } from 'react';
+import {
+  fetchPendingRewards,
+  fetchTotalClaimedRewards,
+} from 'services/web3/protection/rewards';
+import BigNumber from 'bignumber.js';
+import { prettifyNumber } from 'utils/helperFunctions';
+import { useAppSelector } from 'redux/index';
+import { useInterval } from 'hooks/useInterval';
+import { NavLink } from 'react-router-dom';
+
+export const MyRewards = () => {
+  const { account } = useWeb3React();
+  const [claimableRewards, setClaimableRewards] = useState<number | null>(null);
+  const [totalRewards, setTotalRewards] = useState<number | null>(null);
+  const bntPrice = useAppSelector<string | null>(
+    (state) => state.bancor.bntPrice
+  );
+
+  const fetchRewardsData = async (account: string) => {
```

Comment:
minor: can extract all this. Eventually, the component renders the total and claimable rewards and bnt price - it doesn't really have to care about all the fetches and intervals.